stdlib/public/runtime/ErrorObject.mm

```diff
@@ -491,6 +477,19 @@ bool BridgeFn(NSError *, OpaqueValue*, const Metadata *,
     // Native class or ObjC class should be an NSError subclass.
     if (![srcType->getObjCClassObject() isSubclassOfClass: NSErrorClass])
       return false;
+    
+    srcInstance = *reinterpret_cast<NSError * const*>(src);
+    
+    // A _SwiftNativeNSError box can always be unwrapped to cast the value back
+    // out as an Error existential.
+    if (!reinterpret_cast<SwiftError*>(srcInstance)->isPureNSError()) {
```

Comment:
Nitpick: Strictly speaking this is not legal C++, since it's calling a method on a type whose dynamic type may not match the static type.